File: Easy/MIME Type/Python 3/MIME Type.py
﻿import sys

number_of_elements  = int(input())  
number_of_fileNames = int(input()) 

file_extensions_association_array = {}

for i in range(number_of_elements):
    inputs = input().split()
    file_extensions_association_array[inputs[0].lower()] = inputs[1] # file extension and MIME type.
    
for i in range(number_of_fileNames):
    file_name = input()  # One file name per line.
    extension = "";
    
    if (file_name.find(".") != -1): # Checking for a dot
        extension = file_name[file_name.rfind(".") + 1 : len(file_name)].lower() # Including a dot
        
    print(file_extensions_association_array[extension] if file_extensions_association_array.get(extension) else "UNKNOWN")

File: Easy/MIME Type/Java/MIMEType.java
import java.util.*;

class Solution {

    public static void main(String args[]) {
        Scanner in = new Scanner(System.in);

        int numberOfElements = in.nextInt(); // Number of elements which make up the association table.
        in.nextLine();
        int numberOfFileNames = in.nextInt(); // Number Q of file names to be analyzed.
        in.nextLine();

        HashMap<String, String> file_extensions_association_dict = new HashMap<String, String>();

        for (int i = 0; i < numberOfElements; i++) {
            String extension = in.next(); // file extension
            String mimeType = in.next(); // MIME type.
            in.nextLine();

            file_extensions_association_dict.put(extension.toLowerCase(), mimeType);
        }

        for (int i = 0; i < numberOfFileNames; i++) {
            String fileName = in.nextLine(); // One file name per line.
            String extension = "";

            if (fileName.indexOf(".") != -1) // Checking for a dot
                extension = fileName.substring(fileName.lastIndexOf(".") + 1).toLowerCase(); // Including a dot

            System.out.println(file_extensions_association_dict.containsKey(extension) ? file_extensions_association_dict.get(extension) : "UNKNOWN");
        }
    }
}